tabular/src/autogluon/tabular/models/catboost/catboost_model.py

```diff
@@ -178,88 +224,96 @@ def _fit(self, X_train, y_train, X_val=None, y_val=None, time_limit=None, num_gp
+            if train_dir is not None:
+                params_final['train_dir'] = train_dir
+            if params_final['iterations'] > 0:
+                init_model = self.model
+
+                self.model = model_type(
+                    **params_final,
+                )
+
+                # TODO: Strangely, this performs different if clone init_model is sent in than if trained for same total number of iterations. May be able to optimize catboost models further with this
+                self.model.fit(
+                    X_train,
+                    eval_set=eval_set,
+                    verbose=verbose,
+                    early_stopping_rounds=early_stopping_rounds,
+                    # use_best_model=True,
+                    init_model=init_model,
+                )
+
+                self._merge_prev_model(init_model, metric_name)
         else:
+            #TODO: Time limit constraint is not considered in warm starting scenairo. May need to add it
+            self._prev_model = self.model
             params_final = params.copy()
 
-        if train_dir is not None:
-            params_final['train_dir'] = train_dir
-        if params_final['iterations'] > 0:
+            if warm_iter:
+                params_final["iterations"] = warm_iter
+
             self.model = model_type(
                 **params_final,
             )
 
-            # TODO: Strangely, this performs different if clone init_model is sent in than if trained for same total number of iterations. May be able to optimize catboost models further with this
             self.model.fit(
                 X_train,
                 eval_set=eval_set,
                 verbose=verbose,
-                early_stopping_rounds=early_stopping_rounds,
-                # use_best_model=True,
-                init_model=init_model,
+                #early_stopping_rounds=early_stopping_rounds,
+                use_best_model=False,
+                init_model=self._prev_model,
             )
 
-            if init_model is not None:
-                final_model_best_score = self.model.get_best_score()['validation'][metric_name]
-                if self.stopping_metric._optimum > final_model_best_score:
-                    if final_model_best_score > init_model_best_score:
-                        best_iteration = init_model_tree_count + self.model.get_best_iteration()
-                    else:
-                        best_iteration = init_model_best_iteration
-                else:
-                    if final_model_best_score < init_model_best_score:
-                        best_iteration = init_model_tree_count + self.model.get_best_iteration()
-                    else:
-                        best_iteration = init_model_best_iteration
-
-                self.model.shrink(ntree_start=0, ntree_end=best_iteration+1)
-
+            # progress = self._merge_prev_model(self.prev_model, metric_name, params_final["iterations"], X_val=X_val, y_val=y_val)
```

Comment:
Why is this commented out?

Reply:
Currently CatBoost doesn't support early stopping during incremental training. I kept the code since it could be useful for future. But right now it can't be used.